### PR TITLE
Handle the `useAutoItem` option in the URL generator

### LINKED ISSUE #489: UrlGenerator and auto_item disabled
Just thinking about it, can we still disable `auto_items` in Contao 4? And if so, I think the UrlGenerator would ignore that?

### PR DESCRIPTION
**Pull request by leofeyer**
This PR fixes #489.

### CODE REVIEW

File: src/Routing/UrlGenerator.php [hunk 1]
```diff
@@ -142,13 +158,13 @@ function ($matches) use ($alias, &$parameters, $autoItems, &$hasAutoItem) {
                 $value = $parameters[$param];
                 unset($parameters[$param]);
 
-                if (!$hasAutoItem && in_array($param, $autoItems, true)) {
-                    $hasAutoItem = true;
-
-                    return $value;
+                if (!$config->get('useAutoItem') || $hasAutoItem || !in_array($param, $autoItems, true)) {
```
Comment (aschempp) [inline src/Routing/UrlGenerator.php]: Are you sure this is correct? The auto item should only be used if the config flag is true

Comment (leofeyer) [inline src/Routing/UrlGenerator.php]: The if condition now returns `$param.'/'.$value` :)

Comment (aschempp) [inline src/Routing/UrlGenerator.php]: You're right, my fault

**Comment by leofeyer** [comment]
Merged in 6efa96281d5ab20ada7258143f6bdbef858557ac.

**Pull request merged**